docs/user_guide/backend_configuration.rst

```diff
@@ -0,0 +1,286 @@
+      full shape of source array : (3,)
+      full size of source array : 0.00 GB
+
+      buffer shape : (2,)
+      expected RAM usage : 0.00 GB
+
+      chunk shape : (1,)
+      disk space usage per chunk : 0.00 MB
+
+      compression method : Zstd
+      compression options : {'clevel': 3}
+
+
+Interfaces and Converters
+-------------------------
+
+The normal workflow when writing an NWB file using a ``DataInterface`` or ``NWBConverter`` is simple to configure.
+
+The following example uses the :ref:`example data <example_data>` available from the testing repo...
+
+.. code-block:: python
+
+    from datetime import datetime
+
+    from dateutil import tz
+    from neuroconv import ConverterPipe
+    from neuroconv.datainterfaces import SpikeGLXRecordingInterface, PhySortingInterface
+    from neuroconv.tools.nwb_helpers import (
+        make_or_load_nwbfile,
+        get_default_backend_configuration,
+        configure_backend,
+    )
+
+    # Instantiate interfaces and converter
+    ap_interface = SpikeGLXRecordingInterface(
+        file_path=".../spikeglx/Noise4Sam_g0/Noise4Sam_g0_imec0/Noise4Sam_g0_t0.imec0.ap.bin"
+    )
+    phy_interface = PhySortingInterface(
+        folder_path=".../phy/phy_example_0"
+    )
+
+    data_interfaces = [ap_interface, phy_interface]
+    converter = ConverterPipe(data_interfaces=data_interfaces)
+
+    # Fetch available metadata
+    metadata = converter.get_metadata()
+
+    # Create the in-memory NWBFile object and retrieve a default configuration
+    backend="hdf5"
+
+    nwbfile = converter.create_nwbfile(metadata=metadata)
+    backend_configuration = converter.get_default_backend_configuration(
+        nwbfile=nwbfile,
+        backend=backend,
+    )
+
+    # Make any modifications to the configuration in this step, for example...
+    backend_configuration["acquisition/ElectricalSeriesAP/data"].compression_method = "Blosc"
+
+    # Configure and write the NWB file
+    nwbfile_path = "./my_nwbfile_name.nwb"
+    converter.run_conversion(
+        nwbfile_path=nwbfile_path,
+        nwbfile=nwbfile,
+        backend=backend,
+        backend_configuration=backend_configuration,
+    )
+
+.. note::
+
+    If you do not intend to make any alterations to the default configuration for the given backend type, then you can follow the classic workflow...
+
+    .. code-block::
+
+        converter = ConverterPipe(data_interfaces=data_interfaces)
+
+        # Fetch available metadata
+        metadata = converter.get_metadata()
+
+        # Create the in-memory NWBFile object and retrieve a default configuration
+        backend="hdf5"
+
+        # Configure and write the NWB file
+        nwbfile_path = "./my_nwbfile_name.nwb"
+        converter.run_conversion(
+            nwbfile_path=nwbfile_path,
+            nwbfile=nwbfile,
+            backend=backend,
+        )
+
+    and all datasets in the NWB file will automatically use the default configuration!
+
+
+Generic tools
```

Comment:
I can't make sense of this section. What does it provide that the other approaches do not? Maybe we can have that in the first paragraph. Something like:

Maybe you want to do x and that is not possible with the other approaches, to solve your problem you can use the following approach:

Reply:
This section showcases how it can be used in conjunction with non-NeuroConv approaches (any in-memory NWBFile)

The others use the built-in class methods, not the base tool method